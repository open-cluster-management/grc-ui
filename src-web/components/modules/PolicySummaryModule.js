/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import resources from '../../../lib/shared/resources'
import msgs from '../../../nls/platform.properties'
import _ from 'lodash'
import { getAvailableGrcFilters } from '../../../lib/client/filter-helper'

resources(() => {
  require('../../../scss/module-policy-summary.scss')
})

export default class PolicySummaryModule extends React.Component {

  render() {
    const { locale } = this.context
    const title = msgs.get('overview.policy.summary.title', locale)
    const data = this.getModuleData()
    return (
      <div className='module-policy-summary'>
        <div className='card-container-container'>
          <div className='card-title'>
            {title}
          </div>
          <div className='card-container'>
            <div className='card-content'>
              <div className='card-inner-content'>
                {data.summary.map(({count, summaryType}) => {
                  return (
                    <div key={summaryType} className='card-count-type'>
                      <div className='card-count'>
                        {count}
                      </div>
                      <div className='card-type'>
                        <div>
                          {summaryType.toUpperCase()}
                        </div>
                      </div>
                    </div>
                  )
                })}

              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }

  getModuleData = () => {
    const { locale } = this.context
    const { policies, findings } = this.props
    const filters = getAvailableGrcFilters(policies, findings, locale)
    const policySet = new Set()
    const standardsSet = filters.standards.availableSet
    const categorySet = filters.categories.availableSet
    const controlSet = filters.controls.availableSet

    policies.map(policy=>policySet.add(_.get(policy, 'metadata.name', 'unknown')))
    return {
      summary: [
        {count: standardsSet.size, summaryType: msgs.get('overview.policy.summary.standards', locale)},
        {count: categorySet.size, summaryType: msgs.get('overview.policy.summary.categories', locale)},
        {count: policySet.size, summaryType: msgs.get('overview.policy.summary.policies', locale)},
        {count: controlSet.size, summaryType: msgs.get('overview.policy.summary.controls', locale)},
      ],
    }
  }
}

PolicySummaryModule.propTypes = {
  findings: PropTypes.array,
  policies: PropTypes.array,
}
