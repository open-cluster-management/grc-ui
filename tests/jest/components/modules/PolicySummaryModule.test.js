/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { policiesTestingDataSet1 } from './ModuleTestingData'
import PolicySummaryModule from '../../../../src-web/components/modules/PolicySummaryModule'

describe('PolicySummaryModule view', () => {
  const filteredPolicies = policiesTestingDataSet1
  it('renders as expected', () => {
    const component = renderer.create(<BrowserRouter><PolicySummaryModule
      policies={filteredPolicies} />
    </BrowserRouter>)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
