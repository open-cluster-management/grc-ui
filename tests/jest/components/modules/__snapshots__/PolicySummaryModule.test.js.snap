// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicySummaryModule view renders as expected 1`] = `
<div
  className="module-policy-summary"
>
  <div
    className="card-container-container"
  >
    <div
      className="card-title"
    >
      Policy summary
    </div>
    <div
      className="card-container"
    >
      <div
        className="card-content"
      >
        <div
          className="card-inner-content"
        >
          <div
            className="card-count-type"
          >
            <div
              className="card-count"
            >
              3
            </div>
            <div
              className="card-type"
            >
              <div>
                STANDARDS
              </div>
            </div>
          </div>
          <div
            className="card-count-type"
          >
            <div
              className="card-count"
            >
              3
            </div>
            <div
              className="card-type"
            >
              <div>
                CATEGORIES
              </div>
            </div>
          </div>
          <div
            className="card-count-type"
          >
            <div
              className="card-count"
            >
              3
            </div>
            <div
              className="card-type"
            >
              <div>
                POLICIES
              </div>
            </div>
          </div>
          <div
            className="card-count-type"
          >
            <div
              className="card-count"
            >
              3
            </div>
            <div
              className="card-type"
            >
              <div>
                CONTROLS
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
